packages/quantic/force-app/main/default/lwc/quanticSearchInterface/quanticSearchInterface.js

```diff
@@ -33,13 +42,44 @@ export default class QuanticSearchInterface extends LightningElement {
             }
           };
           setEngineOptions(this.engineOptions, CoveoHeadless.buildSearchEngine, this.engineId, this);
-          setInitializedCallback(this.performInitialQuery, this.engineId);
+          setInitializedCallback(this.initialize, this.engineId);
         }
-      })
+      });
     });
   }
 
-  performInitialQuery = (engine) => {
-    engine.executeFirstSearch()
+  disconnectedCallback() {
+    if (this.unsubscribeUrlManager) {
+      this.unsubscribeUrlManager();
+    }
+  }
+ 
+  initialize = (engine) => {
+    if (!this.disableStateInUrl) {
+      this.initUrlManager(engine);
+    }
+    engine.executeFirstSearch();
+  }
+
+  get fragment() {
+    return window.location.hash.slice(1);
+  }
+
+  initUrlManager(engine) {
+    this.urlManager = CoveoHeadless.buildUrlManager(engine, {initialState: {fragment: this.fragment}});
+    this.unsubscribeUrlManager = this.urlManager.subscribe(() => this.updateHash());
+    window.addEventListener('hashchange', this.onHashChange);
+  }
+
+  updateHash() {
+    window.history.pushState(
```

Comment:
:thinking: Should we check that `this.urlManager.state.fragment` is different from the previous one? Does `urlManager` already take care of it?

Reply:
The url manager handles this, it only calls subscribers when one of the params changes